pkg/volume/aws_ebs/attacher.go

```diff
@@ -264,13 +264,14 @@ func (detacher *awsElasticBlockStoreDetacher) Detach(volumeName string, nodeName
 			volumeID, nodeName, err)
 	}
 
+	nodeExists := true
+
 	if err == nil && !attached {
-		// Volume is already detached from node.
-		glog.Infof("detach operation was successful. volume %q is already detached from node %q.", volumeID, nodeName)
-		return nil
+		// it means either node has been deleted or volume is not attached to node
```

Comment:
DiskIsAttached will return not attached without error only if cloudprovider.InstanceNotFound returns true. If the node is deleted from cloudprovider, doesn't mean volume should be automatically detached?

Reply:
no. in AWS even nodes that are shutdown/stopped are removed from node list and volumes are not detached from such nodes.